.github/workflows/ci.yml

```diff
@@ -144,10 +144,95 @@ jobs:
+        uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # v3.3.1
+        with:
+          path: ${{ github.workspace }}/dist
+          key: nginx-kubernetes-gateway-${{ github.run_id }}-${{ github.run_number }}
+
+      - name: Docker Buildx
+        uses: docker/setup-buildx-action@ecf95283f03858871ff00b787d79c419715afc34 # v2.7.0
+      
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@818d4b7b91585d195f67373fd9cb0332e31a7175 # v4.6.0
+        with:
+          images: |
+            name=ghcr.io/nginxinc/nginx-kubernetes-gateway
+          tags: |
+            type=semver,pattern={{version}}
+            type=edge
+            type=ref,event=pr
+            type=ref,event=branch,suffix=-rc,enable=${{ startsWith(github.ref, 'refs/heads/release') }}
+
+      - name: Prepare NKG files
+        run: |
+          nkg_prefix=$(echo ${{ steps.meta.outputs.tags }} | cut -d ":" -f 1)
+          nkg_tag=$(echo ${{ steps.meta.outputs.tags }} | cut -d ":" -f 2)
+          yq -i "with(.spec.template.spec.containers[0]; .image = \"${nkg_prefix}:${nkg_tag}\" | .imagePullPolicy = \"Never\")" deploy/manifests/deployment.yaml
+        shell: bash
+
+      - name: Build Docker Image
+        uses: docker/build-push-action@2eb1c1961a95fc15694676618e422e8ba1d63825 # v4.1.1
+        with:
+          file: build/Dockerfile
+          tags: ${{ steps.meta.outputs.tags }}
+          context: "."
+          target: goreleaser
+          load: true
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          pull: true
+      
+      - name: Build Test Docker Image
+        uses: docker/build-push-action@2eb1c1961a95fc15694676618e422e8ba1d63825 # v4.1.1
+        with:
+          file: conformance/tests/Dockerfile
+          tags: conformance-test-runner:${{ github.sha }}
+          context: "."
+          load: true
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          pull: true
+
+      - name: Deploy Kubernetes
+        id: k8s
+        run: |
+          kind create cluster --image kindest/node:v1.27.1 --kubeconfig kube-${{ github.run_id }}
+          echo "KUBECONFIG=kube-${{ github.run_id }}" >> "$GITHUB_ENV"
+        working-directory: ./conformance
+        shell: bash
+
+      - name: Setup conformance tests
+        run: |
+          nkg_prefix=$(echo ${{ steps.meta.outputs.tags }} | cut -d ":" -f 1)
+          nkg_tag=$(echo ${{ steps.meta.outputs.tags }} | cut -d ":" -f 2)
+          kind load docker-image ${nkg_prefix}:${nkg_tag}
+          make preload-nginx-container
+          make prepare-nkg-dependencies
+          yq '(select(di != 3))' provisioner/provisioner.yaml | kubectl apply -f -
+          yq "(select(.spec.template.spec.containers[].image) | .spec.template.spec.containers[].image=\"${nkg_prefix}:${nkg_tag}\" | .spec.template.spec.containers[].imagePullPolicy = \"Never\")" provisioner/provisioner.yaml | kubectl apply -f -          
```

Comment:
Any reason we can't use the existing make targets for all of this?

Reply:
Only because this is a draft and I had feedback in the other PR - this will be updated